src/languages/php.js

```diff
@@ -12,13 +12,13 @@ Category: common
  * */
 export default function(hljs) {
   const regex = hljs.regex;
-  const IDENT_RE_CORE = '[a-zA-Z0-9_\x7f-\xff]*' +
+  const IDENT_RE_CORE = 'a-zA-Z0-9_\x7f-\xff]*' +
     // negative look-ahead tries to avoid matching patterns that are not
     // Perl at all like $ident$, @ident@, etc.
     '(?![A-Za-z0-9])(?![$]))';
-  const IDENT_RE = regex.concat("([a-zA-Z_\\x7f-\\xff]", IDENT_RE_CORE);
+  const IDENT_RE = regex.concat("([a-zA-Z_\\x7f-\\xff][", IDENT_RE_CORE);
   // Will not detect camelCase classes
-  const PASCAL_CASE_CLASS_NAME_RE = regex.concat("([A-Z]", IDENT_RE_CORE);
+  const PASCAL_CASE_CLASS_NAME_RE = regex.concat("([A-Z][\\\\", IDENT_RE_CORE);
```

Comment:
Please create a second constant (to allow `\`) if you must, but don't split the expressions up this way, this is hard to follow and is likely to increase maintenance in the future.

Reply:
Hmm, it's correct to have a class name preceded by a namespace like: `new Assert\Count()`  
Just fixing what I missed earlier.  
I wanted to eliminate duplicate code. Two almost the same regex might be harder to maintain in the future.

Reply:
If you have another idea, I'm listening, but splitting the character group up and hiding the `[` all over the code, nope. :)

Reply:
A more precise regex would be preferable... because were not really trying to match:

`A\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\`

We're trying to match `Ident\Indent` or some such... it'd be better to do that that the imprecise thing.

Reply:
> A more precise regex would also be better... because were not really trying to match:
> 
> `A\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\`
> 
> We're trying to match `Ident\Ident` or some such... it'd be better to simple do that vs extending the character group like this.

Fixed